tests/v2/validation/workloads/reusable_workload_tests.go

```diff
@@ -0,0 +1,490 @@
+		{"WorkloadDaemonSetTest", WorkloadDaemonSetTest},
+		{"WorkloadCronjobTest", WorkloadCronjobTest},
+		{"WorkloadStatefulsetTest", WorkloadStatefulsetTest},
+		{"WorkloadUpgradeTest", WorkloadUpgradeTest},
+		{"WorkloadPodScaleUpTest", WorkloadPodScaleUpTest},
+		{"WorkloadPodScaleDownTest", WorkloadPodScaleDownTest},
+		{"WorkloadPauseOrchestrationTest", WorkloadPauseOrchestrationTest},
+	}
+
+	return tests
+}
+
+// WorkloadDeploymentTest creates a deployment and validates that it is successfully deployed.
+func WorkloadDeploymentTest(client *rancher.Client, clusterID string) error {
+	log.Info("Creating a new project and namespace")
+	_, namespace, err := projectsapi.CreateProjectAndNamespace(client, clusterID)
+	if err != nil {
+		return err
+	}
+
+	log.Info("Creating a new deployment")
+	_, err = deployment.CreateDeployment(client, clusterID, namespace.Name, 1, "", "", false, false)
+
+	return err
+}
+
+// WorkloadSideKickTest updates a deployment and validates the result.
+func WorkloadSideKickTest(client *rancher.Client, clusterID string) error {
+	log.Info("Creating a new project and namespace")
+	_, namespace, err := projectsapi.CreateProjectAndNamespace(client, clusterID)
+	if err != nil {
+		return err
+	}
+
+	log.Info("Creating a new deployment")
+	createdDeployment, err := deployment.CreateDeployment(client, clusterID, namespace.Name, 1, "", "", false, false)
+	if err != nil {
+		return err
+	}
+
+	log.Info("Waiting for all pods to be running")
+	err = pods.WatchAndWaitPodContainerRunning(client, clusterID, namespace.Name, createdDeployment)
+	if err != nil {
+		return err
+	}
+
+	containerName := namegen.AppendRandomString("update-test-container")
+	newContainerTemplate := workloads.NewContainer(containerName,
+		redisImage,
+		corev1.PullAlways,
+		[]corev1.VolumeMount{},
+		[]corev1.EnvFromSource{},
+		nil,
+		nil,
+		nil,
+	)
+
+	createdDeployment.Spec.Template.Spec.Containers = append(createdDeployment.Spec.Template.Spec.Containers, newContainerTemplate)
+
+	log.Info("Updating image deployment")
+	updatedDeployment, err := deployment.UpdateDeployment(client, clusterID, namespace.Name, createdDeployment)
+	if err != nil {
+		return err
+	}
+
+	log.Info("Waiting for deployment to become active")
+	err = charts.WatchAndWaitDeployments(client, clusterID, namespace.Name, metav1.ListOptions{
+		FieldSelector: "metadata.name=" + updatedDeployment.Name,
+	})
+	if err != nil {
+		return err
+	}
+
+	log.Info("Waiting for all pods to be running")
+	err = pods.WatchAndWaitPodContainerRunning(client, clusterID, namespace.Name, updatedDeployment)
+
+	return err
+}
+
+// WorkloadDaemonSetTest creates a daemonset and validates that it is successfully deployed.
+func WorkloadDaemonSetTest(client *rancher.Client, clusterID string) error {
+	log.Info("Creating a new project and namespace")
+	_, namespace, err := projectsapi.CreateProjectAndNamespace(client, clusterID)
+	if err != nil {
+		return err
+	}
+
+	log.Info("Creating a new deamonset")
```

Comment:
Super nit - this should be `daemonset` and not `deamonset`.

Reply:
Good point.